.github/workflows/snapshot-abi.yaml

```diff
@@ -0,0 +1,177 @@
+# Test ABI versions against snapshot
+#
+#
+
+name: ABI Test Against Snapshot
+"on":
+  schedule:
+    # run daily 20:00 on main branch
+    - cron: '0 20 * * *'
+  push:
+    branches:
+      - prerelease_test
+      - trigger/snapshot-abi
+  pull_request:
+    paths: .github/workflows/snapshot-abi.yaml
+jobs:
+  config:
+    runs-on: ubuntu-latest
+    outputs:
+      pg14_abi_min: ${{ steps.config.outputs.pg14_abi_min }}
+      pg15_abi_min: ${{ steps.config.outputs.pg15_abi_min }}
+      pg16_abi_min: ${{ steps.config.outputs.pg16_abi_min }}
+      pg17_abi_min: ${{ steps.config.outputs.pg17_abi_min }}
+
+    steps:
+    - name: Checkout source code
+      uses: actions/checkout@v4
+    - name: Read configuration
+      id: config
+      run: python .github/gh_config_reader.py
+
+  abi_snapshot_test:
+    name: ABI Snapshot Test PG${{ matrix.pg }}
+    runs-on: ubuntu-latest
+    needs: config
+
+    env:
+      PG_SRC_DIR: pgbuild
+      PG_INSTALL_DIR: postgresql
+      PG_EXTENSIONS: postgres_fdw test_decoding pageinspect pgstattuple
+
+    strategy:
+      fail-fast: false
+      matrix:
+        pg: [ 14, 15, 16, 17 ]
+        include:
+          - pg: 14
+            abi_min: ${{ fromJson(needs.config.outputs.pg14_abi_min) }}
+            pg_extra_args: --enable-debug --enable-cassert --without-llvm
+          - pg: 15
+            abi_min: ${{ fromJson(needs.config.outputs.pg15_abi_min) }}
+            pg_extra_args: --enable-debug --enable-cassert --with-llvm LLVM_CONFIG=llvm-config-14
+          - pg: 16
+            abi_min: ${{ fromJson(needs.config.outputs.pg16_abi_min) }}
+            pg_extra_args: --enable-debug --enable-cassert --with-llvm LLVM_CONFIG=llvm-config-14
+          - pg: 17
+            abi_min: ${{ fromJson(needs.config.outputs.pg17_abi_min) }}
+            pg_extra_args: --enable-debug --enable-cassert --with-llvm LLVM_CONFIG=llvm-config-14
```

Comment:
I guess we can move `--enable-debug --enable-cassert` out of pg_extra_args here. Can we have them all without llvm to  not have the extra config diff or is there merit in testing llvm builds explicitly

Reply:
Our current regression tests are run with the same extra args, so I included them here. I think we can remove the `debug` and `cassert` if we want to test against a 'release` build.

I kept `--with-llvm` too, since it improves query performance -- is there any specific reason to not keep it?